### PR TITLE
Validate public origin against custom domains

### PR DESCRIPTION
**Pull request by IniZio**
ref #4022

### CODE REVIEW

File: pkg/portal/graphql/app_mutation.go [hunk 1]
```diff
@@ -287,11 +287,44 @@ var _ = registerMutationField(
...
+						if domain.Domain == appConfig.HTTP.PublicOrigin {
+							validOrigin = true
+							break
+						}
+					}
+					if !validOrigin {
+						return nil, errors.New("public origin must be one of the available domains")
+					}
+				}
+
```
Comment (louischan-oursky) [inline pkg/portal/graphql/app_mutation.go:327]: - In pkg/portal/appresource/manager.go, inject appHostSuffixes in context
- In pkg/portal/appresource/manager.go, inject domain service in context. The domain service should be able to list custom domains of a project.
- In pkg/lib/config/configsource/resources.go, update AuthgearYAML UpdateResources to use the newly injected values to perform domain validation.

Comment (IniZio) [reply pkg/portal/graphql/app_mutation.go]: Updated 🙏

**Comment by louischan-oursky** [comment]
lint failed and test failed.

File: pkg/lib/config/configsource/resources.go [hunk 2]
```diff
@@ -12,24 +13,39 @@ import (
 	"github.com/authgear/authgear-server/pkg/api/apierrors"
 	"github.com/authgear/authgear-server/pkg/api/model"
 	"github.com/authgear/authgear-server/pkg/lib/config"
+	portalmodel "github.com/authgear/authgear-server/pkg/portal/model"
```
Comment (louischan-oursky) [inline pkg/lib/config/configsource/resources.go:16]: pkg/lib cannot import pkg/portal. The workaround is to move Domain to pkg/api/model.

File: pkg/portal/model/app.go [hunk 3]
```diff
@@ -20,8 +20,13 @@ type App struct {
 	Context *config.AppContext
 }
 
+type DescriptedPath struct {
```
Comment (louischan-oursky) [inline pkg/portal/model/app.go:23]: Move DescriptedPath to pkg/util/resource/descriptor.go

File: pkg/lib/config/configsource/resources.go (hunk 2 above)
Comment (IniZio) [reply pkg/lib/config/configsource/resources.go]: Updated path and removed original workaround

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**